server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/filter/JMAPFilteringTest.java

```diff
@@ -0,0 +1,3393 @@
+                            Rule.builder()
+                                .id(RULE_ID_1)
+                                .name(RULE_NAME_1)
+                                .condition(Rule.Condition.of(Rule.Condition.Field.TO, Rule.Condition.Comparator.NOT_CONTAINS, "Frédéric_MARTIN_alternative_name"))
+                                .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of(testSystem.getFredMartinInbox().serialize()))))
+                                .build()));
+
+                FakeMail mail = FakeMail.builder()
+                        .sender(SENDER_1)
+                        .recipient(FRED_MARTIN)
+                        .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                            .addHeader("to", "sender2 <sender2@james.org>")
+                            .addHeader("to", "=?UTF-8?B?RnLDqWTDqXJpYyBNQVJUSU4=?= <fred.martin@linagora.com>,\r\n" +
+                                " sender1 <sender1@james.org>"))
+                        .build();
+
+                testSystem.getJmapFiltering().service(mail);
+
+                assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + FRED_MARTIN_USERNAME))
+                        .isEqualTo(FRED_MARTIN_INBOX);
+            }
+        }
+
+        @Nested
+        class ToFilteringWithExactlyEqualsCommandRule implements WithExactlyEqualsCommandRule {
+
+            @Test
+            @Override
+            public void mailDirectiveShouldBeSetWhenAddressExactlyEqualsRuleValue(JMAPFilteringTestSystem testSystem) throws Exception {
+                testSystem.getFilteringManagement().defineRulesForUser(User.fromUsername(RECIPIENT_1_USERNAME),
+                        ImmutableList.of(
+                            Rule.builder()
+                                .id(RULE_ID_1)
+                                .name(RULE_NAME_1)
+                                .condition(Rule.Condition.of(Rule.Condition.Field.TO, Rule.Condition.Comparator.EXACTLY_EQUALS, "sender1@james.org"))
+                                .action(Rule.Action.of(Rule.Action.AppendInMailboxes.withMailboxIds(ImmutableList.of(testSystem.getRecipient1MailboxId().serialize()))))
+                                .build()));
+
+                FakeMail mail = FakeMail.builder()
+                        .sender(SENDER_2)
+                        .recipients(RECIPIENT_1, RECIPIENT_2)
+                        .mimeMessage(MimeMessageBuilder.mimeMessageBuilder()
+                            .addHeader("to", "sender1 <sender1@james.org>"))
+                        .build();
+
+                testSystem.getJmapFiltering().service(mail);
+
+                assertThat(mail.getAttribute(DELIVERY_PATH_PREFIX + RECIPIENT_1_USERNAME))
+                        .isEqualTo(RECIPIENT_1_MAILBOX_1);
+            }
+
+            @Override
+            public void mailDirectiveShouldBeSetWhenPersonalPersonalExactlyEqualsRuleValue(JMAPFilteringTestSystem testS
```

Comment:
Show me a test with addresses like `.addHeader("from", "sender1@james.org")` which are also valid...

Reply:
Show me a test with addresses like `.addHeader("from", "")`

Reply:
Show me a test with addresses like `.addHeader("from", "GA BOU ZO MEU")`

I'm expecting `Rule(user, FROM, CONTAINS, "BOU"` to match

Goal: exercize your address parsing error handling logic